### PR TITLE
[第5組]refactor: Making variable more readable

### PR DESCRIPTION
**Pull request by Diego0923**

### CODE REVIEW

**Comment by EdwardKYC** [comment]
同學你太晚交作業了歐 也沒有註解你改了什麼

**Comment by Steven0811** [comment]
There are still many unreadable names not being refactored like p1, p2, l, m, etc. Magic numbers are not extracted.

**Comment by Rogue8464** [comment]
大部分的修改有提升程式可讀性，但部分命名像是is_user_password_correct,get_logged_in_user過於模糊，有些function命名只是把get改成fetch，無法提升程式可讀性

**Comment by percykugood** [comment]
有些詞可以寫得更好例如362行的cnt,491行的m,368行的uid也可以考慮改成user_id

**Comment by Yi-Cheng-Wang** [comment]
**Evaluation:**
- Degree of variable refactoring is low
- Variable renaming is acceptable
- Code layout is tidy
- Commits are organized
- The PR does not explain any changes

**Suggestions:**
- Magic numbers seem to be left unaddressed; for example, the value `1892dhianiandowqd0n` on line 290 could be passed in a better way.
- The variable `bgs` on line 77 could have a more meaningful name.
- Variables like `p` and `p2` on lines 161 and 162, along with many others, have not been changed.
- The variable `ai` on line 202 could be expressed in a better way.
- The parameters passed on line 279 should be optimized.
- The variable `expires_delta` on line 294 is not easily understandable and should be changed.
- The variable `m` on lines 460 and 470 should be changed.
- The variable `u` on lines 484, 486, 490, 494, and 501 should be changed.
- The variable `id2` on line 515 could have a better name.

**Comment by mlcp29th-hi** [comment]
Hello,
1. 建議命名時若有更好的選擇，最好盡量避免 data/info/result 等形式，否則以旁人的角度較不容易立刻理解 `news_data`、`news_info`、`result`等的意義／差別，畢竟變數本身即資料，`_data` 等後綴對於理解其意義的幫助有限。

3. `fetch_news_info(search_term, is_initial_fetch=False)` 的函式名稱清楚描述此函式會執行 fetch 的任務，或許參數 `is_initial_fetch` 可以維持 `is_initial` 即可。

4. 部分程式在處理新聞資料時會用 `…_news_…`／`…_article_...`／`..._news_article_...` 命名，建議可統一一種做法以維持一致性。

5. `delete_stmt`、`insert_stmt` 可以考慮改成 `delete_statement`、`insert_statement` 使人們討論此程式時便於發音，較利於溝通。其他以縮寫命名的變數如 `bgs` 也可考慮換個更容易理解的名稱。

File: backend/main.py [hunk 1]
```diff
@@ -478,34 +478,34 @@ async def news_summary(
...
-            user_news_association_table.c.news_articles_id == n_id,
+            user_news_association_table.c.news_articles_id == username_id,
             user_news_association_table.c.user_id == u_id,
         )
     ).scalar()
 
     if existing_upvote:
         delete_stmt = delete(user_news_association_table).where(
-            user_news_association_table.c.news_articles_id == n_id,
+            user_news_association_table.c.news_articles_id == username_id,
```
Comment (taro0513) [inline backend/main.py:500]: This is not the ID of the username, and this could mislead the meaning of this variable

**Review by taro0513 — COMMENTED** [review]
The commit message is clear, making it easy to understand the purpose of the changes. However, separating it into more detailed parts would improve clarity.

**Pull request merged**